app/controllers/application_controller.rb

```diff
@@ -17,6 +18,12 @@ def user_not_authorized
     redirect_to(authenticated_root_path)
   end
 
+  def check_subdomain
+    if user_signed_in? && (request.subdomain != current_user.organization.subdomain)
+      redirect_to authenticated_root_url(subdomain: current_user.organization.subdomain), allow_other_host: true
+    end
```

Comment:
might be wrong but what if I am a visitor and want to check one random company's wishlists, will I be redirected?

Reply:
I haven't take that into account! Thanks for pointing it out @AGorinam !
So the condition is they can visit other company's wishlist that are public?

Reply:
It feels like we should create helpers for these conditionals, we are starting to have many, but we will do that in a different pull request :)